src/app/Library/CrudPanel/Traits/HasViewNamespaces.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Backpack\CRUD\app\Library\CrudPanel\Traits;
+
+trait HasViewNamespaces
+{
+    private $viewNamespaces = [];
+
+    /**
+     * Return the namespaces stored for the given domain. (fields, filters, buttons etc).
+     *
+     * @param  string  $domain
+     * @return array
+     */
+    public function getViewNamespacesFor(string $domain)
+    {
+        return $this->viewNamespaces[$domain] ?? [];
+    }
+
+    /**
+     * Return the resulting array after merging the base namespaces
+     * with the ones stored for the given domain.
+     *
+     * @param  string  $domain
+     * @param  null|string  $configNamespace
+     * @return array
+     */
+    public function getAllViewNamespacesFor(string $domain, string $configNamespace = null)
+    {
+        $configNamespace ??= $this->getConfigNamespaceFor($domain);
+
+        return array_unique(array_merge(config($configNamespace) ?? [], $this->getViewNamespacesFor($domain)));
+    }
```

Comment:
I like all method here except this one... this one... something made me a bit jittery... and I couldn't find why... but now I think I got it:
- (1) seems like a duplicate of `getViewNamespacesFor()`... because it is, it's only got an extra parameter;
- (2) having `all` in the name is weird, when getViewNamespacesFor()` kind of also gets all of them;
- (3) its second parameter was confusing to me, what's a config namespace? a "configured" namespace? a "configuration" namespace? a namespace "configuration"?

So here's what I propose... 🥁🥁🥁

### (B) fallback

Hear me out! In the off chance that this returns nothing... we let the dev specify what the return should be. BAM. Wouldn't that be easier to use and understand?

```suggestion
    public function getViewNamespacesFor(string $domain, array $fallback = [])
    {
        $fallback ??= $this->getDefaultNamespaceFor($domain);

        return array_unique(array_merge($fallback, $this->viewNamespaces[$domain] ?? []));
    }
```

Reply:
Realistically though... this second parameter will only be used ONCE: 

```php
app('crud')->getAllViewNamespacesFor('widgets', ['backpack.base.component_view_namespaces.widgets']));
```

and it kind of pollutes a beautiful method here... so we could just aswell...

### (C) lose the second parameter

Then in that one place where we need the fallback we do:


```php
app('crud')->getAllViewNamespacesFor('widgets') ?? ['backpack.base.component_view_namespaces.widgets'];
```

😅 right?

Reply:
I think it'd just be cleaner and easier to do option C, right?

Reply:
Oh riiiight. So what we actually need for Widgets is something like this, then?

### (D) lose the second parameter, but add a function for that one place where we need it

```php
array_unique(array_merge(['backpack.base.component_view_namespaces.widgets'], app('crud')->getAllViewNamespacesFor('widgets')))
```

I guess we can create a special function for that, that is only used in that one spot. Difficult part is naming it... perhaps...

```php
public function getViewNamespacesWithFallback(string $domain, array $fallback) {
    return array_unique(array_merge($fallback, $this->getAllViewNamespacesFor($domain)));
}
```

Would that be the cleanest?